server/modules/arenaImport/service/arenaImport/jobs/recordsImportJob.js

```diff
@@ -70,20 +71,47 @@ export default class RecordsImportJob extends Job {
       }
 
       const recordToStore = this.prepareRecordToStore(record)
-      await RecordManager.insertRecord(this.user, surveyId, recordToStore, true, this.tx)
+
+      await this.insertOrSkipRecord({ record: recordToStore, nodesBatchPersister })
+
+      this.incrementProcessedItems()
+    })
+
+    await nodesBatchPersister.flush()
+  }
+
+  async fetchRecordSummary({ recordUuid }) {
+    const { surveyId } = this.context
+    const { list } = await RecordManager.fetchRecordsSummaryBySurveyId({ surveyId, recordUuid }, this.tx)
+    return list[0]
+  }
+
+  async insertOrSkipRecord({ record, nodesBatchPersister }) {
+    const { survey, surveyId } = this.context
+
+    const recordUuid = Record.getUuid(record)
+
+    const existingRecord = await this.fetchRecordSummary({ surveyId, recordUuid })
+
+    if (!existingRecord || DateUtils.isAfter(Record.getDateModified(record), Record.getDateModified(existingRecord))) {
+      if (existingRecord) {
```

Comment:
If(!existing){if (existing) ...

Reply:
eheheh, it is like "insert the record if"
1) the record does not exist already
2) or the record being imported has more recent changes
in one of these 2 cases we delete the existing record only if it exists